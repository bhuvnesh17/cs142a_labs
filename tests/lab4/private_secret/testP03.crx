// test operator associativity

func main() : void {
  var a : int;
  var b : int;
  var c : int;
  var d : int;
  var e : int;
  var f : int;
  var g : int;

  let g = a + b - c + d - e - f;
  let g = a * b / c * d / e / f;
  let g = a and b or c and d or e or f;
}
