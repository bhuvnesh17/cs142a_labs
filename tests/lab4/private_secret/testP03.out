ast.DeclarationList(3,1)
  ast.FunctionDefinition(3,1)[Symbol(main), []]
    ast.StatementList(4,3)
      ast.VariableDeclaration(4,3)[Symbol(a)]
      ast.VariableDeclaration(5,3)[Symbol(b)]
      ast.VariableDeclaration(6,3)[Symbol(c)]
      ast.VariableDeclaration(7,3)[Symbol(d)]
      ast.VariableDeclaration(8,3)[Symbol(e)]
      ast.VariableDeclaration(9,3)[Symbol(f)]
      ast.VariableDeclaration(10,3)[Symbol(g)]
      ast.Assignment(12,3)
        ast.AddressOf(12,7)[g]
        ast.Subtraction(12,29)
          ast.Subtraction(12,25)
            ast.Addition(12,21)
              ast.Subtraction(12,17)
                ast.Addition(12,13)
                  ast.Dereference(12,11)
                    ast.AddressOf(12,11)[a]
                  ast.Dereference(12,15)
                    ast.AddressOf(12,15)[b]
                ast.Dereference(12,19)
                  ast.AddressOf(12,19)[c]
              ast.Dereference(12,23)
                ast.AddressOf(12,23)[d]
            ast.Dereference(12,27)
              ast.AddressOf(12,27)[e]
          ast.Dereference(12,31)
            ast.AddressOf(12,31)[f]
      ast.Assignment(13,3)
        ast.AddressOf(13,7)[g]
        ast.Division(13,29)
          ast.Division(13,25)
            ast.Multiplication(13,21)
              ast.Division(13,17)
                ast.Multiplication(13,13)
                  ast.Dereference(13,11)
                    ast.AddressOf(13,11)[a]
                  ast.Dereference(13,15)
                    ast.AddressOf(13,15)[b]
                ast.Dereference(13,19)
                  ast.AddressOf(13,19)[c]
              ast.Dereference(13,23)
                ast.AddressOf(13,23)[d]
            ast.Dereference(13,27)
              ast.AddressOf(13,27)[e]
          ast.Dereference(13,31)
            ast.AddressOf(13,31)[f]
      ast.Assignment(14,3)
        ast.AddressOf(14,7)[g]
        ast.LogicalOr(14,35)
          ast.LogicalOr(14,30)
            ast.LogicalOr(14,19)
              ast.LogicalAnd(14,13)
                ast.Dereference(14,11)
                  ast.AddressOf(14,11)[a]
                ast.Dereference(14,17)
                  ast.AddressOf(14,17)[b]
              ast.LogicalAnd(14,24)
                ast.Dereference(14,22)
                  ast.AddressOf(14,22)[c]
                ast.Dereference(14,28)
                  ast.AddressOf(14,28)[d]
            ast.Dereference(14,33)
              ast.AddressOf(14,33)[e]
          ast.Dereference(14,38)
            ast.AddressOf(14,38)[f]

